src/utilities/report-utils.ts

```diff
@@ -67,34 +67,44 @@ export const getReportUrl = (questionRefId?: string) => {
     result += "&runKey=" + runKey
             + "&activity=" + resourceUrl
             + "&resourceUrl=" + resourceUrl;
-  }
-  else {
+    return result;
+  } else {
     // We know this is a IPortalData because there is no runKey
     const portalData = getPortalData() as IPortalData;
-    const classInfoUrl = portalData?.portalJWT?.class_info_url;
-    const authDomainUrl = classInfoUrl?.split("/api")[0];
-    const offeringBaseUrl = classInfoUrl?.split("/classes")[0]+"/offerings/";
-    const offeringId = portalData?.offering.id;
-    const offeringUrl = encodeURIComponent(offeringBaseUrl + offeringId);
-    const studentId = portalData?.platformUserId;
+    // Handles logged in teachers that are previewing a non-assigned resource or are in teacher-edition mode
+    if (!portalData?.offering.id) {
+      return null;
+    } else {
+      const classInfoUrl = portalData?.portalJWT?.class_info_url;
+      const authDomainUrl = classInfoUrl?.split("/api")[0];
+      const offeringBaseUrl = classInfoUrl?.split("/classes")[0]+"/offerings/";
+      const offeringId = portalData?.offering.id;
```

Comment:
In a PR I just merged Doug just changed this to be `portalData?.offering?.id`.  
I think you should merge master into this PR and choose to keep it the way it is here. 
The `if (!portalData.offering.id) {` along with the changes to the tests should take care of this. 

If there isn't an offering we'd want to throw an error so we can track down where things went wrong.